packages/sidecar/src/mcp/parsing.ts

```diff
@@ -0,0 +1,66 @@
+type RawEventContext = {
```

Comment:
Might be good to add a top-level comment about where this code came from and our future plans of splitting this out.